src/test/java/com/googleinterns/zoomtube/servlets/DiscussionServletTest.java

```diff
@@ -0,0 +1,205 @@
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.users.User;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.appengine.tools.development.testing.LocalUserServiceTestConfig;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.reflect.TypeToken;
+import com.googleinterns.zoomtube.data.Comment;
+import com.ryanharter.auto.value.gson.GenerateTypeAdapter;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringReader;
+import java.io.StringWriter;
+import java.lang.reflect.Type;
+import java.time.Clock;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+public class DiscussionServletTest {
+  private static final int LECTURE_ID = 1;
+  private static final String LECTURE_ID_STR = "1";
+  private static final String TEST_COMMENT_CONTENT = "Test content";
+  private static final String DEFAULT_EMAIL = "test@example.com";
+  private static final String AUTH_DOMAIN = "example.com";
+
+  @Rule public final MockitoRule mockito = MockitoJUnit.rule();
+
+  @Mock private HttpServletRequest request;
+  @Mock private HttpServletResponse response;
+  private DiscussionServlet servlet;
+  private LocalServiceTestHelper testServices = new LocalServiceTestHelper(
+      new LocalUserServiceTestConfig(), new LocalDatastoreServiceTestConfig());
+  private DatastoreService datastore;
+
+  @Before
+  public void setUp() throws ServletException {
+    testServices.setUp();
+    testServices.setEnvEmail(DEFAULT_EMAIL);
+    testServices.setEnvAuthDomain(AUTH_DOMAIN);
+    servlet = new DiscussionServlet();
+    servlet.init();
+    datastore = DatastoreServiceFactory.getDatastoreService();
+  }
+
+  @After
+  public void cleanUp() {
+    testServices.tearDown();
+  }
+
+  @Test
+  public void doPost_loggedOut_postForbidden() throws ServletException, IOException {
+    testServices.setEnvIsLoggedIn(false);
+    when(request.getParameter(DiscussionServlet.PARAM_LECTURE)).thenReturn(LECTURE_ID_STR);
+
+    servlet.doPost(request, response);
+
+    verify(response).sendError(eq(HttpServletResponse.SC_FORBIDDEN), anyString());
+    assertThat(datastore.prepare(new Query(Comment.ENTITY_KIND)).countEntities(withLimit(1)))
+        .isEqualTo(0);
+  }
+
+  @Test
+  public void doPost_storesCommentWithProperties_noParent() throws ServletException, IOException {
```

Comment:
This test name confuses me. What are you testing? Seems you are testing 2 things, storing comments and not having any parents. Should it be split into 2 tests?

Reply:
I check every property except the parent property, and test the parent property in another test.  Maybe I should combine them?

Reply:
I moved things around so it's more clear what I'm testing